cluster_management/src/main/java/com/pinterest/rocksplicator/ConfigGenerator.java

```diff
@@ -237,4 +242,37 @@ private boolean updateDisabledHosts() {
     disabledHosts = latestDisabledInstances;
     return true;
   }
+
+  /**
+   * filter out workflows and jobs from helix resources, only return db resources from ideal states
+   *
+   * db_resource and wf are guaranteed to be unique, bcz wf is treated as "resource" in helix,
+   * no db_resource and wf are allowed with same name during db_resource or wf creation
+   */
+  public List<String> filterOutWfAndJobs(List<String> resources) {
+    List<String> db_resources = new ArrayList<>();
+
+    Map<String, WorkflowConfig> workflowConfigMap = (driver == null ? null : driver.getWorkflows());
+
+    if (driver != null && workflowConfigMap != null && !workflowConfigMap.isEmpty()) {
+      Set<String> workflows = workflowConfigMap.keySet();
+      for (String resource : resources) {
+        boolean isWfOrJob = false;
+        for (String workflow : workflows) {
+          if (!workflow.isEmpty() && resource.startsWith(workflow)) {
```

Comment:
How do we guarantee that no db resource will start with any workflow name?

Is there any function/tag/something in helix zk that can help us tell if a resource is a regular db resource or workflow resource directly?

Reply:
We have planned to enforce duplication check (specifically, prefix duplication check) when new db resource or workflow addition. This can prevent no db resource and workflow/jobs will have same name. 

I can't find any existing method to retrieve db only resources from zk. I will ask in helix community to confirm this.

Reply:
Will replacing startswith() with equals() work?

The implementation of this function can be refactored/simplified a little bit.

Reply:
I got some feedback from helix team, will re-implement the filtering.